resources/kubernetes/operator/helm/values.yaml

```diff
@@ -219,11 +219,30 @@ collectors:
     # Configure the pods resources to control CPU and memory usage.
     resources:
       limits:
-        cpu: 100m
-        memory: 500Mi
+        cpu: 1500m
+        memory: 1500Mi
       requests:
         cpu: 100m
         memory: 500Mi
+    env:
+      - name: ELASTIC_AGENT_OTEL
+        value: '"true"'
+      - name: ELASTIC_ENDPOINT
+        valueFrom:
+          secretKeyRef:
+            name: elastic-secret-otel
+            key: elastic_endpoint
+      - name: ELASTIC_API_KEY
+        valueFrom:
+          secretKeyRef:
+            name: elastic-secret-otel
+            key: elastic_api_key
+      - name: GOMAXPROCS
+        valueFrom:
+          resourceFieldRef:
+            resource: limits.cpu
+      - name: GOMEMLIMIT
+        value: "1025MiB"
```

Comment:
Maybe we can write a script to keep this 90% (or lower if we want to go lower) of the memory limit? Will be a bit easier to manage and update in future.

Reply:
Definitely, that would be a better setting. As discussed, we can help with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30289 moving it forward.